### PR TITLE
Allow after fn's that can be called with no args

### PR DESCRIPTION
**Pull request by dpsutton**
Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [x] You've added tests to cover your change(s)
- [ ] All tests are passing
- [ ] The new code is not generating reflection warnings
- [x] You've updated the readme (if adding/changing middleware)

Thanks!

Not sure how to run all tests actually. I was able to run the clj ones with `lein with-profile +test-clj test` but i'm not sure if there's an easy way to run them all.

I'm also not sure how to ensure that I am not generating any reflection warnings but I don't think that I've added anything that would, but I'm happy to check this before merging.

### CODE REVIEW

**Comment by Malabarba** [comment]
I don't think we have a test for reflection warnings, so don't worry about it. 

The cljs test was failing for you, but I think it was a false failure, so I restarted the test for you.

**Comment by bbatsov** [comment]
You'll have to rebase this on top of the current `master`.

File: test/clj/cider/nrepl/middleware/refresh_test.clj [hunk 1]
```diff
@@ -61,14 +65,22 @@
       (is (:error response)))))
 
 (deftest after-arg
```
Comment (bbatsov) [inline test/clj/cider/nrepl/middleware/refresh_test.clj]: Maybe we should use `after-arg-fn` as our term in the tests. "After arg" sounds a bit odd.

**Comment by bbatsov** [comment]
@dpsutton ping :-)

**Comment by dpsutton** [comment]
Thanks for the ping, sorry for the necessity :) Rebased and test names updated.

**Comment by bbatsov** [comment]
👍

**Pull request merged**